pkg/api/defaults.go

```diff
@@ -701,11 +701,11 @@ func (p *Properties) setAgentProfileDefaults(isUpgrade, isScale bool, cloudName
 func (p *Properties) setWindowsProfileDefaults(isUpgrade, isScale bool) {
 	windowsProfile := p.WindowsProfile
 	if !isUpgrade && !isScale {
-		if windowsProfile.WindowsPublisher == "" {
-			windowsProfile.WindowsPublisher = DefaultWindowsPublisher
-		}
 
 		if p.IsAzureStackCloud() {
+			if windowsProfile.WindowsPublisher == "" {
+				windowsProfile.WindowsPublisher = DefaultWindowsPublisher
```

Comment:
@rjaini - do you want to change the AzureStack default in a separate PR or this one?

Reply:
yes. As the VHD issue is not resolve yet.